src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs

```diff
@@ -34,24 +34,32 @@ protected BaseOtlpHttpExportClient(OtlpExporterOptions options, HttpClient httpC
     internal IReadOnlyDictionary<string, string> Headers { get; }
 
     /// <inheritdoc/>
-    public bool SendExportRequest(TRequest request, CancellationToken cancellationToken = default)
+    public ExportClientResponse SendExportRequest(TRequest request, CancellationToken cancellationToken = default)
     {
+        DateTime deadline = DateTime.UtcNow.AddMilliseconds(this.HttpClient.Timeout.TotalMilliseconds);
         try
         {
             using var httpRequest = this.CreateHttpRequest(request);
 
             using var httpResponse = this.SendHttpRequest(httpRequest, cancellationToken);
 
-            httpResponse?.EnsureSuccessStatusCode();
+            try
+            {
+                httpResponse.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException ex)
```

Comment:
Avoid this try-catch by moving the response declaration outside of the parent try block.